src/domain/flow-system/src/entities/flow/flow_id.rs

```diff
@@ -37,7 +35,6 @@ impl From<FlowID> for u64 {
 
 /////////////////////////////////////////////////////////////////////////////////////////
 
-pub type FlowIDStream<'a> =
-    std::pin::Pin<Box<dyn tokio_stream::Stream<Item = Result<FlowID, InternalError>> + Send + 'a>>;
+pub type FlowIDStream<'a> = std::pin::Pin<Box<dyn tokio_stream::Stream<Item = FlowID> + Send + 'a>>;
```

Comment:
Why did this stream became infallible? Aren't we touching a database for queries like `get_all_flow_ids_by_dataset`?

Reply:
My original thought was that only query as a whole coudl fail, but it appears I am wrong after investigating Diesel API.
There are no built-in async streams there , but, let's assume we would use load by iterators.
The return type of such queries is:

```
type RowIter<'conn>: Iterator<Item = QueryResult<U>>...
```

and query result is defined as:

```
enum QueryResult<T> {
    Ok(T),
    Err([Error]),
}
```

So, iterator/stream version should really have result in the items, while queries returning individual items immediately would return `Result`.